source/neuropod/bindings/c/np_tensor.cc

```diff
@@ -18,6 +18,13 @@ limitations under the License.
 #include "neuropod/bindings/c/np_tensor_internal.h"
 #include "neuropod/internal/neuropod_tensor_raw_data_access.hh"
 
+void NP_GetDims(const NP_NeuropodTensor *tensor, size_t *num_dims, const int64_t **dims)
+{
+    auto dimsCollection = tensor->tensor->as_tensor()->get_dims();
```

Comment:
Use `auto &` here.

The `dims` pointer you're setting below is not pointing to a valid location on the heap (since `dimsCollection` will be destroyed when this function returns and the pointer returned by `data()` will no longer be valid).

The test below happens to pass because the data isn't being overwritten before you check against the expected value.

(Also, nit: snake case instead of camel case for variable names - `tensor_dims` or something instead of `dimsCollection`)

Reply:
Yes, I always forget that auto never resolves to reference (with few exceptions)!

Do you mean, because I missed auto& it returns pointer to local - agree. With auto& it will point to what get_dims returns that return reference to tensor's internal buffer and so it should be fine, right?

Reply:
Correct, the data pointed to by `dims` is valid as long as the tensor is valid

Reply:
Fixed.

Reply:
> Do you mean, because I missed auto& it returns pointer to local - agree.

One clarification - `dims` would point to memory on the heap either way (because the data in a `vector` is on the heap).

The difference is by using `auto &` the returned data is actively managed by a vector that is still "alive" (the `dims_` vector inside NeuropodTensor).

By using `auto`, you're returning a pointer to data that was allocated and managed by `dimsCollection` but is now unused from the system's perspective.